lib/src/delegates/asset_picker_builder_delegate.dart

```diff
@@ -170,9 +189,7 @@ abstract class AssetPickerBuilderDelegate<Asset, Path> {
 
   /// Whether the delegate should build the special item.
   /// 是否需要构建自定义 item
-  bool get shouldBuildSpecialItem =>
-      specialItemPosition != SpecialItemPosition.none &&
-      specialItemBuilder != null;
+  bool get shouldBuildSpecialItem => specialItems.isNotEmpty;
```

Comment:
any suggestion for this?

Reply:
Maybe we can get rid of it.

Reply:
If `shouldBuildSpecialItem` is removed, then it will always show loading indicator even there are special item to be displayed. Below is the current code which depends on `shouldBuildSpecialItem`.

```dart
 final bool shouldDisplayAssets =
              p.hasAssetsToDisplay || shouldBuildSpecialItem;
          return AnimatedSwitcher(
            duration: switchingPathDuration,
            child: shouldDisplayAssets
                ? Stack(
                    children: <Widget>[
                      RepaintBoundary(
                        child: Column(
                          children: <Widget>[
                            Expanded(child: assetsGridBuilder(context)),
                            bottomActionBar(context),
                          ],
                        ),
                      ),
                      pathEntityListBackdrop(context),
                      pathEntityListWidget(context),
                    ],
                  )
                : loadingIndicator(context),
```

Reply:
`shouldBuildSpecialItem` has been removed, this is because `assetsGridBuilder` has handled the total asset count checking to display either loadingIndicator or list.

```dart
 final List<SpecialItemModel> specialItemModels = specialItems
            .map((item) {
              final specialItem = item.builder?.call(
                context,
                wrapper?.path,
                totalCount,
                permissionNotifier.value,
              );
              if (specialItem != null) {
                return SpecialItemModel(
                  position: item.position,
                  item: specialItem,
                );
              }
              return null;
            })
            .whereType<SpecialItemModel>()
            .toList();

        totalCount += specialItemModels.length;

        if (totalCount == 0 && specialItemModels.isEmpty) {
          return loadingIndicator(context);
        }
```